modules/grub.sh

```diff
@@ -155,24 +110,23 @@ EOF
     sudo rm -rf work/boot/EFI/systemd
     sudo rm -rf work/boot/NvVars
     for empty_dir in $(
-                     cd work/boot find . -maxdepth 1 -type d -empty
+                     cd work/boot && find . -maxdepth 1 -type d -empty
```

Comment:
I would consider squashing this whole commit (the refactor) with the previous one. It's a bit confusing to review the first one, since it is in fact wrong, and due to this change, it won't even work. When viewed with `-w`, the two commits together look pretty good!